### ISSUE TITLE
Cursor skips on HTML entities 

### ISSUE THREAD

**Issue by darthneel** [froala/react-froala-wysiwyg#61]
### Issue 

Any HTML entity causes the cursor to skip back to the beginning

![cursorskip](https://user-images.githubusercontent.com/5307165/35171557-136cd3a2-fd19-11e7-82da-e2815369e5b3.gif)

### Details

The underlying HTML is  `see that skiP? Cursor skips on &quot;`

It appears to be the `&quot;` character that breaks it. The same thing happens with `&rsquo;`. The odd part is that if you edit the HTML directly and change `&rsquo;` to `&#39;`, it works perfectly. 

I've tried including the `entities` plugin and defining some entities in my config but had no success there.

**Comment by darthneel** [comment]
@stefanneculai No, i cannot reproduce this on the website. On top of using `react-froala-wysiwyg`, we have another app that uses the non-React version and I also cannot reproduce this bug there. This issue seems to be specific to the React version.

Some additional observations: 

If i do the following: 
```javascript
<FroalaEditor 
  model='she said "hello "world"' 
/>
```

The editor itself seems to be converting the model to `she said &quot;hello "world&quot;'`. Is there a way for me to manually convert the `&quot;` characters to `&#34;` characters? Switching to the numerical character entity seems to solve the issue.

Thanks in advance for any help you can provide! This is turning into a big issue with our users and I appreciate any help I can get.

**Comment by DeyLak** [comment]
I faced the same behaviour, but for me, it seems to have different reason. If I type text fast - it is ok, but when I make a pause, it skips to the first position. I also was not able to reproduce this on demo :(. I think, it somehow can be related to dispatching a bunch of redux actions on model change. 
I was able to fix that by linking model of the editor to the component state(also leaved it dublicated in redux store)

**Comment by stefanneculai** [comment]
@darthneel @DeyLak what browser and OS are you using?

**Comment by DeyLak** [comment]
@stefanneculai Mac OS, Chrome 63.0.3239.1320 (But I think, here can be some variety, our users are also experiencing this problem)

**Comment by chiunhau** [comment]
+1 for the issue

**Comment by darthneel** [comment]
@stefanneculai At this point, I would settle for being able to turn off the `entities` plugin and just deal with HTML entities myself. 

I've tried deleting the `entities.js` plugin file and also passing `{entities: ''}` into the config options to disable it, but neither seem to work. In both cases `$.FroalaEditor.PLUGINS` still contains the `entities` key and the plugin logic is clearly being executed. 

Can you provide some guidance on how to properly disable the `entities` plugin?

**Comment by m-andrew-albright** [comment]
I believe I'm experiencing this same issue.  If it's not the same, it's very similar (and I'll be happy to open a separate issue if it is not the same)

For me, I am using the npm package `sanitize-html` to strip out malicious content, in case a malicious user attempts to XSS another user who can edit their content.

```jsx
<FroalaEditor
  model={sanitizeUserHtml(this.props.html)}
  tag="textarea"
  config={defaultFroalaConfig}
  onModelChange={this.props.updateHtml}
/>
```

I would like to rely on froala to disable any XSS vectors, but currently at least the `<img onerror="alert('MALICIOUS JS RUNNING')" src="#" />` vector works, so I am attempting to mitigate this myself.  Unfortunately it introduces unusable UX, because the cursor reverts to the beginning.

Please let me know if I should open a separate issue for this.

**Comment by darthneel** [comment]
@stefanneculai Any update on my comment from 6 days ago? This is affecting our customers and I need to get it resolved ASAP.

**Comment by stefanneculai** [comment]
@darthneel @DeyLak @chiunhau it looks the problem comes from the fact that you're using two-way binding and you make some changes to the HTML output (like `sanitizeUserHtml`). When the new HTML is different, it is set back to the editor so that the model and editor content are in sync. As the input / output is pure HTML it cannot keep the cursor position and the cursor goes back to the initial position. If you would like to make any postprocessing to the HTML output, we'd recommend doing this in the `html.get` event: https://www.froala.com/wysiwyg-editor/docs/events#html.get. Thus, the model and editor content wouldn't be out of sync.

Sorry I kept you guys waiting on this, however, there was no clear way of reproducing it and none of the demos we provide (on our website or here in the React repo) have this problem.

**Comment by aleccool213** [comment]
Anyone have an example of using `html.get` to get this working with React?
